File: app/lib/util/device.dart
import 'dart:io';

import 'package:device_info_plus/device_info_plus.dart';
import 'package:flutter_udid/flutter_udid.dart';

class DeviceInfo {
  DeviceInfo({
    required this.osVersion,
    required this.manufacturer,
    required this.model,
    required this.deviceId,
  });

  final String osVersion;
  final String manufacturer;
  final String model;
  final String deviceId;
}

Future<DeviceInfo> getDeviceInfo() async {
  String deviceId = await FlutterUdid.consistentUdid;
  if (Platform.isIOS) {
    final iosInfo = await DeviceInfoPlugin().iosInfo;
    return DeviceInfo(
      osVersion: iosInfo.systemVersion ?? "",
      manufacturer: "Apple",
      model: iosInfo.name ?? "",
      deviceId: deviceId,
    );
  } else {
    // it must be Android if not iOS
    final androidInfo = await DeviceInfoPlugin().androidInfo;
    return DeviceInfo(
      osVersion: androidInfo.version.release!,
      manufacturer: androidInfo.manufacturer,
      model: androidInfo.model,
      deviceId: deviceId,
    );
  }
}

File: invy_api/device/device.go
package device

import "github.com/hashicorp/go-version"

type Device struct {
	AppVersion     *version.Version // e.g. 1.0.0
	AppBuildNumber int              // e.g. 153
	Manufacturer   string           // e.g. Apple
	Platform       string           // e.g. ios
	OSVersion      string           // e.g. 16.2
	Model          string           // e.g. iPhone 14
}
